lib/src/scanner.dart

```diff
@@ -224,30 +348,84 @@ class NgScanner {
       _state = NgScannerState.scanElementDecorator;
       return scanBeforeElementDecorator();
     }
-    if (_current.type == NgSimpleTokenType.voidCloseTag ||
-        _current.type == NgSimpleTokenType.tagEnd) {
+    if (type == NgSimpleTokenType.voidCloseTag ||
+        type == NgSimpleTokenType.tagEnd) {
       return scanElementEndOpen();
     }
-    return handleError();
+
+    if (type == NgSimpleTokenType.openBracket ||
+        type == NgSimpleTokenType.openParen ||
+        type == NgSimpleTokenType.openBanana ||
+        type == NgSimpleTokenType.hash ||
+        type == NgSimpleTokenType.star ||
+        type == NgSimpleTokenType.equalSign ||
+        type == NgSimpleTokenType.closeBracket ||
+        type == NgSimpleTokenType.closeParen ||
+        type == NgSimpleTokenType.closeBanana ||
+        type == NgSimpleTokenType.doubleQuote ||
+        type == NgSimpleTokenType.singleQuote) {
+      return handleError(
+        NgParserWarningCode.AFTER_ELEMENT_IDENTIFIER_OPEN_NEED_WHITESPACE,
+        _lastToken.offset,
+        _lastToken.length,
+      );
+    }
+
+    if (type == NgSimpleTokenType.commentBegin ||
+        type == NgSimpleTokenType.openTagStart ||
+        type == NgSimpleTokenType.closeTagStart ||
+        type == NgSimpleTokenType.EOF) {
+      return handleError(
+        NgParserWarningCode.AFTER_ELEMENT_IDENTIFIER,
+        _lastToken.offset,
+        _lastToken.length,
+      );
+    }
+
+    return handleError(
```

Comment:
Agreed. I've actually thought of using a table too that maps out errors as well as correct/standard path. I'll introduce that into a separate PR.